urdf2mjcf/convert.py

```diff
@@ -226,6 +232,52 @@ def add_default(root: ET.Element, metadata: ConversionMetadata) -> None:
             attrib["ctrlrange"] = f"-{joint_param.actuatorfrc} {joint_param.actuatorfrc}"
         ET.SubElement(sub_default, "motor", attrib=attrib)
 
+    actuator_types = set()
+    for joint_name, joint_metadata in joint_name_to_metadata.items():
+        if joint_metadata is None:
+            raise ValueError(f"Missing metadata for joint: {joint_name}")
+        if not isinstance(joint_metadata, dict):
+            raise ValueError(f"Metadata for joint {joint_name} is not a dictionary")
+        actuator_types.add(joint_metadata["actuator_type"])
+        logger.info(f"Joint {joint_name} uses actuator type: {joint_metadata['actuator_type']}")
+
+    logger.info(f"Found {len(actuator_types)} actuator types in metadata: {actuator_types}")
+
+    # Create default classes for each actuator type
+    for actuator_type in actuator_types:
+        if actuator_type is None:
+            raise ValueError("Actuator type is None")
+
+        sub_default = ET.SubElement(robot_default, "default", attrib={"class": str(actuator_type)})
+
+        joint_attrib = {}
+        motor_attrib = {}
+
+        if actuator_type not in actuator_type_to_metadata:
+            raise ValueError(f"Missing actuator type metadata for {actuator_type}")
+        actuator_data = actuator_type_to_metadata[str(actuator_type)]
+        logger.info(f"Creating class for {actuator_type} with params: {actuator_data}")
+
+        # Set parameters for joint
+        if hasattr(actuator_data, "armature"):
+            joint_attrib["armature"] = str(actuator_data.armature)
+
+        if hasattr(actuator_data, "frictionloss"):
+            joint_attrib["frictionloss"] = str(actuator_data.frictionloss)
+
+        if hasattr(actuator_data, "damping"):
+            joint_attrib["damping"] = str(actuator_data.damping)
+
+        # Set parameters for motor
+        if hasattr(actuator_data, "max_torque"):
+            joint_attrib["actuatorfrcrange"] = f"-{actuator_data.max_torque} {actuator_data.max_torque}"
+            motor_attrib["ctrlrange"] = f"-{actuator_data.max_torque} {actuator_data.max_torque}"
```

Comment:
i strongly prefer not using `hasattr` here. we should make the actuator data conform to some type instead, with these fields as optional